### ISSUE TITLE
Cant get Kubernetes task's log when using CeleryKubernetesExecutor

### ISSUE THREAD

**Issue by KulykDmytro** [apache/airflow#21548]
### Apache Airflow version

2.2.3 (latest released)

### What happened

Recently we're switched from `KubernetesExecutor` to `CeleryKubernetesExecutor` and faced with strange behaviour
Tasks which are sent to `kubernetes` queue unable to provide logs in `webserver` while executing
```
*** Falling back to local log
*** Log file does not exist: /opt/airflow/logs/sfmc_sync/notification_center_send_log/2022-02-12T07:00:00+00:00/6.log
*** Fetching from: http://10.47.204.47:8793/log/sfmc_sync/notification_center_send_log/2022-02-12T07:00:00+00:00/6.log
*** Failed to fetch log file from worker. [Errno 113] No route to host
```
webserver not even tried to reach tasks log over k8s api (like it happened when using `KubernetesExecutor`)

### What you expected to happen

When task is executed on `kubernetes` queue (despite of executor chosen), logs from container's stdout should be provided to webserver upon request

### How to reproduce

set executor to `CeleryKubernetesExecutor`
set default queue to `celery`
set additional queue for `kubernetes`
send task for execution to `queue='kubernetes'`

### Operating System

Linux 5.4.156-83.273.amzn2.x86_64

### Versions of Apache Airflow Providers

standard ones for official image

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

EKS deployment

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by arkadiusz-bach** [comment]
This is a bug, it is trying to access Celery logs because of the following line:
https://github.com/apache/airflow/blob/710d6997035a5e050367db013ae6847cc90dc51a/airflow/utils/log/file_task_handler.py#L143

It should also check whether executor is equal to 'CeleryKubernetesExecutor' and the task queue is 'kubernetes'(queue value should be taken from config)

Once you have got that fixed, log will be empty in the running state, because it is trying to read container logs and by default tasks logs are written to files.

You will need to either add stdout handler for `airflow.task` logger in logging config or do it like I did, I just replaced the read namespaced pod with kubernetes exec command, which just executes `tail -n 100 log_file_path' in the task container in order to get last 100 lines of log. 

As for the failed or succeeded tasks, it depends whether you have got remote_logging enabled, if you do then log will be taken from remote location otherwise it will be empty, because it seems that you are not using the same volumes for webserver and tasks containers and once the task container is gone then logs are gone as well.

You can just mount the same volume for logs directory on both webserver and tasks containers, it will work, but if you are not using the same volumes(or volumes at all) for logs then that is the case.

**Comment by KulykDmytro** [comment]
@arkadiusz-bach i believe you've mentioned how to fix an issue in #11629

**Comment by potiuk** [comment]
> @arkadiusz-bach i believe you've mentioned how to fix an issue in #11629

This is adifferent story. I think this one is a real issue that needs to be handled.

**Comment by amdubovoy** [comment]
Any updates on the issue? The same thing is happening with LocalKubernetesExecutor

**Comment by uranusjr** [comment]
Feel free to draft a pull request to work toward a fix.

**Comment by potiuk** [comment]
Added a "good first issue" label - maybe someone will pick it up @amdubovoy - this is an easy way to become one of 2.300 contributors to Airflow...

**Comment by joonsan-kim-moloco** [comment]
> This is a bug, it is trying to access Celery logs because of the following line:
> 
> https://github.com/apache/airflow/blob/710d6997035a5e050367db013ae6847cc90dc51a/airflow/utils/log/file_task_handler.py#L143
> 
> It should also check whether executor is equal to 'CeleryKubernetesExecutor' and the task queue is 'kubernetes'(queue value should be taken from config)
> 
> Once you have got that fixed, log will be empty in the running state, because it is trying to read container logs and by default tasks logs are written to files.
> 
> You will need to either add stdout handler for `airflow.task` logger in logging config or do it like I did, I just replaced the `read_namespaced_pod_log` with kubernetes exec command, which just executes `tail -n 100 log_file_path' in the task container in order to get last 100 lines of log.
> 
> As for the failed or succeeded tasks, it depends whether you have got remote_logging enabled, if you do then log will be taken from remote location otherwise it will be empty, because it seems that you are not using the same volumes for webserver and tasks containers and once the task container is gone then logs are gone as well.
> 
> You can just mount the same volume for logs directory on both webserver and tasks containers, it will work, but if you are not using the same volumes(or volumes at all) for logs then that is the case.

Hi team, did it resolved on Airflow 2.5.x?
I'm running `CeleryKubernetesExecutor` on Airflow 2.4.3, and arrived here for same problem.
I found that the pointed line `airflow/utils/log/file_task_handler.py#L143` is gone in main branch.

**Comment by potiuk** [comment]
There is  a big change in file task handler coming in 2.6. Whether this is solved along the way - hard to say  - but once 2.6.0rc1 is out - I think the best thing you can do @joonsan-kim-moloco is to get the RC (release candidate) for a spin, migrate on your test system and see if your problem is solved. Just subscribe to the devlist https://airflow.apache.org/community/ , watch for the announcement when the 2.6.0rc1 is out (we do not know yet) and help the community (and yourself to see if the problem is solved there). This is the best contribution back you can make (in return for the free software you have) I think.

**Comment by potiuk** [comment]
The beta version of 2.6.0 is out, and RC1 is coming so if you could check @joonsan-kim-moloco if that is solved that would be great

**Comment by joonsan-kim-moloco** [comment]
> The beta version of 2.6.0 is out, and RC1 is coming so if you could check @joonsan-kim-moloco if that is solved that would be great

Somehow it's been resolved by 2.4.3->2.5.1
Unfortunately other dependencies block building docker image with newer Airflow

**Comment by potiuk** [comment]
> Unfortunately other dependencies block building docker image with newer Airflow

I guess you need to upgrade them though. 

I am marking then that one as closed. We can re-open that one if needs be.

**Comment by thispejo** [comment]
Hey everyone,

Ran into an old problem again in version 2.8 that we thought was fixed back in 2.6.3. We're having trouble getting logs from Kubernetes tasks when using the CeleryKubernetesExecutor. It was all good in 2.6.3, but now in 2.8, it's not working again. Anyone else facing this? Any tips or fixes would be awesome!

Thanks!

**Comment by potiuk** [comment]
> Ran into an old problem again in version 2.8 that we thought was fixed back in 2.6.3. We're having trouble getting logs from Kubernetes tasks when using the CeleryKubernetesExecutor. It was all good in 2.6.3, but now in 2.8, it's not working again. Anyone else facing this? Any tips or fixes would be awesome!

It's likely another issue. I suggest -rather than commenting on closed one. open a new one and describe the logs and circumstances. The truth is that this is issue is closed, it has been reported as fixed, so the circumstances that led to this one are no longer describing how to reproduce it. If you want to  get help on that one you should open a new issue with - hopefully - enough details for someone to be able to help you - either diagnose and fix it or help you with your configuration @thispejo
